celements-model/src/main/java/com/celements/model/object/restriction/FieldPresentRestriction.java

```diff
@@ -0,0 +1,52 @@
+package com.celements.model.object.restriction;
+
+import static com.google.common.base.Preconditions.*;
+
+import java.util.Objects;
+
+import javax.annotation.concurrent.Immutable;
+import javax.validation.constraints.NotNull;
+
+import com.celements.model.classes.fields.ClassField;
+import com.celements.model.object.ObjectBridge;
+
+@Immutable
+public class FieldPresentRestriction<O, T> extends ClassRestriction<O> {
+
+  private final ClassField<T> field;
+
+  public FieldPresentRestriction(@NotNull ObjectBridge<?, O> bridge, @NotNull ClassField<T> field) {
+    super(bridge, field.getClassDef().getClassReference());
+    this.field = checkNotNull(field);
+  }
+
+  public ClassField<T> getField() {
+    return field;
+  }
+
+  @Override
+  public boolean apply(@NotNull O obj) {
+    return super.apply(obj) && getBridge().getObjectFieldAccessor().getValue(obj,
+        getField()).isPresent();
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(super.hashCode(), getField());
```

Comment:
a `FieldPresentRestriction` and `FieldAbsentRestriction` with the same field would have the same hashcode. please add  `this.getClass()` in the `ClassRestriction#hashCode` to prevent this.